### PR TITLE
fix(protocol): change custom headers to Map<String,String>

### LINKED ISSUE #2656: CustomHeaders of JobRecord/ActivatedJob should be of type Map<String,String>
Currently, the custom headers of the `JobRecord` / `ActivatedJob` are of type `Map<String, Object>`. However, the values are always of type String. The custom headers are defined in the BPMN as Strings without any type information.

In order to make the API more clear, we should change the type to `Map<String,String>`.

### PR DESCRIPTION
**Pull request by miguelpires**
closes #2656

### CODE REVIEW

File: protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/MsgPackConverter.java [hunk 1]
```diff
@@ -161,6 +161,19 @@ private static void convert(
...
   }
 
+  public static Map<String, String> convertToStringMap(DirectBuffer buffer) {
+    final byte[] msgpackBytes = BufferUtil.bufferAsArray(buffer);
+    final byte[] jsonBytes = convertToJsonBytes(new ByteArrayInputStream(msgpackBytes));
+
+    final TypeReference<HashMap<String, String>> typeRef =
+        new TypeReference<HashMap<String, String>>() {};
+    try {
+      return JSON_OBJECT_MAPPER.readValue(jsonBytes, typeRef);
```
Comment (Zelldon) [inline protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/MsgPackConverter.java]: Maybe we can create a new method out of that? Which takes the buffer and a typeref?

**Comment by MiguelPires** [comment]
Hey @Zelldon, I refactored the map conversion methods

**Review by ChrisKujawa — APPROVED** [review]

**Comment by Zelldon** [comment]
We need to adjust the GO client sorry

**Comment by MiguelPires** [comment]
Sorry, what needs to be changed in the client? The job tests seem to be working

**Comment by Zelldon** [comment]
https://github.com/zeebe-io/zeebe/blob/develop/clients/go/entities/job.go#L35 for exaample should take a map of string to string

**Comment by Zelldon** [comment]
I think there a some problems with the go tests

**Comment by MiguelPires** [comment]
I think it's fixed now. If it's fine with you, I'll squash and merge

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**